server/data/data-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryUrlStore.java

```diff
@@ -19,13 +19,13 @@
 
 package org.apache.james.mailrepository.api;
 
-import java.util.Set;
+import java.util.stream.Stream;
 
 public interface MailRepositoryUrlStore {
 
     void add(MailRepositoryUrl url);
 
-    Set<MailRepositoryUrl> list();
+    Stream<MailRepositoryUrl> list();
```

Comment:
How do you then expose the fact that there are no duplicates?

Reply:
We can name it "Stream<> listDistincts()"

Reply:
so many occasions to promote vavr : in vavr you have many types of persistent collections, Set being one of them. And no need to handle Stream complexity because you can `{flat,}map` any collection with changing it into a Stream first.

Reply:
it seems maybe you finally found a good reason to introduce vavr :)

Reply:
or maybe I can put comments in every open reviews to demonstrate good uses